tc/core/polyhedral/cuda/codegen.cc

```diff
@@ -681,6 +697,24 @@ size_t& nAstNodes() {
   return n;
 }
 
+// Collect ids of tensors that are only read on the Scop.
+std::unordered_set<isl::id, isl::IslIdIslHash> gatherReadOnlySet(
+    const MappedScop& mscop) {
+  auto root = mscop.schedule();
+  const auto& scop = mscop.scop();
+  std::unordered_set<isl::id, isl::IslIdIslHash> readOnlySet;
+
+  if (mscop.useReadOnlyCache) {
```

Comment:
I tend to prefer return early instead.
That is,
```
if (!mscop.useReadOnlyCache) {
  return readOnlySet;
}
```
although I guess in this case the body is small enough